timesketch/ui/static/js/controllers.js

```diff
@@ -104,21 +98,17 @@ timesketch.controller('TsExploreCtrl', function($scope, $http) {
         };
 
         $scope.saveView = function() {
-            var params = {
-                name: $scope.view_name,
-                query: $scope.query,
-                filter: $scope.filter
-            };
-            // TODO: Refactor out to a service.
-            $http.post('/api/v1/sketches/' + $scope.sketch_id + '/views/', params).success(function(data) {
-                var id = data.objects[0].id;
-                var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
-                window.location.href = view_url;
+            timesketchApi.saveView(
+                $scope.sketch_id, $scope.view_name, $scope.query, $scope.filter)
+                .success(function(data) {
+                    var id = data.objects[0].id;
+                    var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
+                    window.location.href = view_url;
             });
```

Comment:
Great idea! I added a interceptor for response failures that broadcasts http-error. This is picked up by a directive that shows a butterbar telling something went wrong.

I will revisit this in the future to make this more elegant.